examples/write_dir.rs

```diff
@@ -98,22 +99,31 @@ where
         } else if !name.as_os_str().is_empty() {
             // Only if not root! Avoids path spec / warning
             // and mapname conversion failed error on unzip
-            println!("adding dir {path:?} as {name:?} ...");
-            #[allow(deprecated)]
-            zip.add_directory_from_path(name, options.clone())?;
+            println!("adding dir {path_as_string:?} as {name:?} ...");
+            zip.add_directory(path_as_string, options.clone())?;
         }
     }
     zip.finish()?;
     Ok(())
 }
-
-fn doit(
-    src_dir: &str,
-    dst_file: &str,
-    method: zip_next::CompressionMethod,
-) -> zip_next::result::ZipResult<()> {
+fn path_to_string(path: &Path) -> anyhow::Result<String> {
+    let mut path_str = String::new();
+    for component in path.components() {
+        if let std::path::Component::Normal(os_str) = component {
+            if !path_str.is_empty() {
+                path_str.push('/');
+            }
+            let os_str_converted = os_str
+                .to_str()
+                .with_context(|| format!("Non UTF-8 Path {:?}", os_str))?;
+            path_str.push_str(os_str_converted);
+        }
+    }
+    Ok(path_str)
+}
+fn doit(src_dir: &str, dst_file: &str, method: zip_next::CompressionMethod) -> anyhow::Result<()> {
     if !Path::new(src_dir).is_dir() {
-        return Err(ZipError::FileNotFound);
+        return Err(ZipError::FileNotFound.into());
```

Comment:
std::io::ErrorKind::NotADirectory would be more helpful.

Reply:
That is nightly only